### DISCUSSION TITLE
How to make the gpx scheme "recognizable"?

### DISCUSSION THREAD

**Discussion by DavidePatria** [OpenTracksApp/OpenTracks#1605] [General]
I have been using opentracks for 1,5 years now and I have a lot of tracks recorded using it. My necessities are changing and I would like to upload the files to runalyze, but I encounter a problem with the gpx format that is not recognized as of type "running", which causes problems.
I tried looking contemporaneously to a runnerup gpx and an opentracks one to try and adjust the latter to get it recognised as run by runalyze but without success.
I would then like to know: what is the difference and how can I make them equal? I could not figure it out by reverse engineering, which might have been foolish to think in the first place.

The objective would be to have a simple script that I can use to convert the tracks, since I intend to keep using opentracks, at least for trekking.
In this case I am also curious about what tag is used to define the activity type, since I cannot see any in the runnerup gpx, therefore I assume that, at least in this case, the clue the parser finds is elsewhere

**Comment by pstorch** [comment]
Do you have an example export from runnerup for us to look at?

**Reply by DavidePatria** [reply]
only two points with changed location. correctly recognised as run by runalyze, altough 110km long given the random points

```
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
<gpx version="1.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.topografix.com/GPX/1/1" xsi:schemaLocation="http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd" xmlns:gpxtpx="http://www.garmin.com/xmlschemas/TrackPointExtension/v1" creator="RunnerUp FP3">
  <metadata>
    <time>2023-06-09T16:21:42Z</time>
  </metadata>
  <trk>
    <name>RunnerUp-Running-2023-06-09T16:21:42Z</name>
    <desc></desc><!-- State change: 1 2023-06-09T16:21:43Z-->
    <trkseg>
      <trkpt lon="7.0000" lat="41.000000">
        <ele>78.00452</ele>
        <time>2023-06-09T16:21:44Z</time>
        <extensions>
          <gpxtpx:TrackPointExtension>
            <accuracy>4.288</accuracy>
            <speed>0.0</speed>
            <sat>12</sat>
          </gpxtpx:TrackPointExtension>
        </extensions>
      </trkpt>
	  <trkpt lon="7.0000" lat="40.000000">
        <ele>77.80084</ele>
        <time>2023-06-09T16:21:45Z</time>
        <extensions>
          <gpxtpx:TrackPointExtension>
            <accuracy>3.216</accuracy>
            <bearing>178.6</bearing>
            <speed>0.24</speed>
            <sat>12</sat>
          </gpxtpx:TrackPointExtension>
        </extensions>
      </trkpt>
    </trkseg>
  </trk>
</gpx>
```

**Reply by pstorch** [reply]
Hmm, it's only in the track name "RunnerUp-Running-*".
Can you try renaming your track in OpenTracks with this prefix?

**Reply by DavidePatria** [reply]
you know that you are right?
changing from
```
<type><![CDATA[corsa]]></type>
```
to
```
<type><![CDATA[running]]></type>
```
is enough for the track to be recognised as running. I should try with biking and hiking too.
may I suggest to keep the activity type tags in english rather than translating them in the gpx? this seems to be superfluous, but I understand it would need another layer of translation or something to store in english and show in custom language. For now I will switch the language to english and stick with it

**Reply by DavidePatria** [reply]
so, I explored the principle a little bit more and that applies to any kind of activity: the presence of a word that can be linked to the type of activity makes it recognisable. Thanks

**Reply by dennisguse** [reply]
I did some research and created an issue for this #1608.
Best guess, we should not translate the `<type>` and this it.
